### PR TITLE
Feature/#248 merge all the changes

### PR DESCRIPTION
**Pull request by vgenev**
=============================== Coverage summary ===============================
Statements   : 46.73% ( 1120/2397 )
Branches     : 31.51% ( 138/438 )
Functions    : 27.68% ( 147/531 )
Lines        : 47.69% ( 1115/2338 )
================================================================================

1..326
# tests 326
# pass  326

Integration tests summary:
=====================
1..130
# tests 130
# pass  130

# ok

transfer models coverage: 
====================
image

participant coverage: 
================
image

extension & ilp model coverage:
=========================
image

### CODE REVIEW

File: testPI2/unit/domain/transfer/ilp/index.test.js [hunk 1]
```diff
@@ -1,168 +1,197 @@
...
- optionally within square brackets <email>.
- * Gates Foundation
- - Name Surname <name.surname@gatesfoundation.com>
- * Valentin Genev <valentin.genev@modusbox.com>
- * Rajiv Mothilal <rajiv.mothilal@modusbox.com>
- * Miguel de Barros <miguel.debarros@modusbox.com>
- * Nikolay Anastasov <nikolay.anastasov@modusbox.com>
- --------------
- ******/
+// /*****
```
Comment (rmothilal) [inline testPI2/unit/domain/transfer/ilp/index.test.js]: Why is this file still here if everything is commented out?

File: testPI2/integration/helpers/transferTestHelper.js [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+'use strict'
+
+const TransferPreparationModule = require('./transfer')
+const TransferStatePreparationHelper = require('./transferState')
+const StateChangeModel = require('../../../src/domain/transfer/models/transferStateChanges')
+const ExtensionModel = require('../../../src/models/extensions')
+const IlpModel = require('../../../src/models/ilp')
+const TransferModel = require('../../../src/domain/transfer/models/transfers-read-model')
+
+// const preparedData = {
```
Comment (rmothilal) [inline testPI2/integration/helpers/transferTestHelper.js]: does this commented out code need to be here? if so why?

File: src/models/ilp.js [hunk 3]
```diff
@@ -1,42 +1,17 @@
-/*****
```
Comment (rmothilal) [inline src/models/ilp.js]: why is the license agreement removed?

File: src/models/ilp.js [hunk 4]
```diff
@@ -51,6 +26,26 @@ exports.saveIlp = async (ilp) => {
 //   }
 // }
 
+// exports.getByTransferId = async (transferId) => {
```
Comment (rmothilal) [inline src/models/ilp.js]: any reason the commented out code is here? for reference purpose?

File: src/models/ilp.js [hunk 5]
```diff
@@ -61,15 +56,19 @@ exports.getByTransferId = async (transferId) => {
...
-exports.getByIlpId = async (ilpId) => {
-  try {
-    return await Db.ilp.findOne({ ilpId: ilpId })
-      // .innerJoin('transfer', 'transfer.transferId', 'ilp.transferId')
-      // .where('expirationDate', '>', `${Time.getCurrentUTCTimeInMilliseconds()}`) // or maybe ${Moment.utc().toISOString()}
-  } catch (err) {
-    throw new Error(err.message)
-  }
-}
+// exports.update = async (ilp, payload) => {
```
Comment (rmothilal) [inline src/models/ilp.js]: any reason the commented out code is here? for reference purpose?

File: src/domain/transfer/models/transfers-read-model.js [hunk 6]
```diff
@@ -1,15 +1,15 @@
...
-const Moment = require('moment')
+// const Moment = require('moment')
 const Db = require('../../../db')
-const TransferState = require('../state')
+// const TransferState = require('../state')
 const extensionModel = require('../../../models/extensions')
 const Logger = require('@mojaloop/central-services-shared').Logger
 
-const findExpired = (expiresAt) => {
+/* const findExpired = (expiresAt) => {
```
Comment (rmothilal) [inline src/domain/transfer/models/transfers-read-model.js]: any reason the commented out code is here? for reference purpose?

File: src/models/ilp.js [hunk 7]
```diff
@@ -1,42 +1,17 @@
...
- their names indented and be marked with a '-'. Email address can be added
- optionally within square brackets <email>.
- * Gates Foundation
- - Name Surname <name.surname@gatesfoundation.com>
- * Valentin Genev <valentin.genev@modusbox.com>
- * Rajiv Mothilal <rajiv.mothilal@modusbox.com>
- * Miguel de Barros <miguel.debarros@modusbox.com>
- --------------
- ******/
-
```
[the remark this answers is not in the document]
Comment (vgenev) [reply src/models/ilp.js]: a totally non-intentional mistake. Or it wasn't there the file I started with. there were 2 ILP models, in the end we tested one of them and the handlers were using the other one. I will put it in right away.

File: src/models/ilp.js (hunk 3 above)
Comment (vgenev) [reply src/models/ilp.js]: a totally non-intentional mistake. Or it wasn't there the file I started with. there were 2 ILP models, in the end we tested one of them and the handlers were using the other one. I will put it in right away.

File: testPI2/integration/domain/transfer/ilp.test.js [hunk 8]
```diff
@@ -33,7 +33,7 @@ const Db = require('../../../../src/db/index')
 const Logger = require('@mojaloop/central-services-shared').Logger
 const Config = require('../../../../src/lib/config')
 // const Model = require('../../../../src/domain/transfer/models/ilp-model')
-const Service = require('../../../../src/domain/transfer/ilp')
+const Service = require('../../../../src/mo')
```
Comment (rmothilal) [inline testPI2/integration/domain/transfer/ilp.test.js]: I don't think this import is correct. please have a look

File: testPI2/unit/domain/transfer/ilp/model.test.js [hunk 9]
```diff
@@ -132,3 +218,48 @@ Test('Ilp model', async (ilpTest) => {
     }
   })
 })
+
+//   await ilpTest.test('update false ilp', async (assert) => {
```
Comment (rmothilal) [inline testPI2/unit/domain/transfer/ilp/model.test.js]: please remove commented out code

File: src/domain/transfer/models/transfers-read-model.js [hunk 10]
```diff
@@ -1,16 +1,35 @@
+/*****
```
Comment (ggrg) [inline src/domain/transfer/models/transfers-read-model.js]: Please, rename filename and all references to transfer-read-model.js
Currently the test filename does not match (transfer-read-model.test.js)

File: testPI2/integration/helpers/testProducer.js [hunk 11]
```diff
@@ -50,7 +50,25 @@ const fulfil = {
   }
 }
 
-const messageProtocol = {
+// const fulfil = {
```
Comment (ggrg) [inline testPI2/integration/helpers/testProducer.js]: Please, remove the commented out code. It's no longer needed.

File: testPI2/integration/helpers/testProducer.js [hunk 12]
```diff
@@ -62,7 +80,7 @@ const messageProtocol = {
   metadata: {
     event: {
       id: Uuid(),
-      type: 'prepare',
+      type: 'transfer',
```
Comment (ggrg) [inline testPI2/integration/helpers/testProducer.js]: This should be changed back to 'prepare'

**Pull request merged**